content/raw/chat/username_avatar.md

```diff
@@ -0,0 +1,132 @@
+Here we introduce support for users to set their own display name and introduce at least 3 mechaniscs to minimize the risk of impersonation attacks. The first mechanism is the use of deterministic sequence of emojis to uniquely identity a profile, the second mechanism is the use of a deterministic sequence of colors around the profile to differentiate it from other profiles which effectively replaces the 3 word usernames and identicons with a similar deterministic identifier, the third mechanism involves preventing duplicate displaynames from joining the same community. There is also an identity verification mechanism that is used to ensure that the user is who they say they are described in the Identity Verifcation Spec.
+
+## Definitions
+
+- 3 word usernames - A 3 word username is a deterministic sequence of words that uniquely identifies a profile. see [spec](https://specs.status.im/spec/2#3-word-pseudonym--whisperwaku-key-fingerprint)
+- display name - A username choosen by the user to be displayed in the chat.
+- local username - An alias for a profile choosen by the User. Only visible to the user.
+- verified identity - A profile that went through the Identity Verification process. See Identity Verification Spec.
+- profile emblem - an icon next to a profile that is used to qualify that profile.
+- emoji hash -  A deterministic sequence of emojis that uniquely identifies a profile.
+- identicon ring - A deterministic sequence of colors around a profile avar that is used to differentiate it from other profiles.
+
+## Designs
+
+- [designs](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=1837%3A335126)
+- [identicon ring](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=1837%3A335130)
+
+## Use Cases
+
+### Setting up a profile
+
+prequesites: User is on onboarding process or has already set up a profile and has choosen to rename their username.
+
+1. User fills their intended display name
+2. User presses Save or Next
+3. System saves the new display name
+3.1 alternative: System warns user that the display name is already taken in one of the communities the user is part of
+3.2 alternative: System warns user that the display name is invalid
+
+### Joining a community
+
+1. User selects a community to join
+2. System detects that another user with the same display name has already joined that community
+3. System warns user that the display name is already taken in that community
+4. System displays conflicting profiles display name, avatar, identicon ring, emoji hash and chat key
+5. User selects "Change Your display name"
+6. User fills in a new display name
+7. Systems adds user to the community
+7.1 alternative: System warns user that the display name is already taken in one of the communities the user is part of
+7.2 alternative: System warns user that the display name is invalid
+
+## Functional Requirements
```

Comment:
The specs does not indicate what should happen when an user receives a message from an account that does not have a display name, which could happen when the author of a message has not updated their mobile or desktop app.
What should be displayed then? the three word alias?

Reply:
Maybe it's a good idea to mention that existing users that update the app to a new version that supports this spec must undergo a process to set their display name before being able to chat

Reply:
Agree on both points. For backwards compatibility, I'd say it would make sense to display the 3 word name